src/State/Layer.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of Raven.
+ *
+ * (c) Sentry Team
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sentry\State;
+
+use Sentry\ClientInterface;
+
+/**
+ * This class holds a pair of client and scope instances for each element in the
+ * stack of a {@see Hub}.
+ *
+ * @internal
+ */
+final class Layer
+{
+    /**
+     * @var ClientInterface|null The client held by this layer
+     */
+    private $client;
+
+    /**
+     * @var Scope The scope held by this layer
+     */
+    private $scope;
+
+    /**
+     * Constructor.
+     *
+     * @param ClientInterface|null $client The client held by this layer
+     * @param Scope                $scope  The scope held by this layer
+     */
+    public function __construct(?ClientInterface $client, Scope $scope)
```

Comment:
What's the reason behind having a layer without a client? To disable sending events?

Reply:
I'm not sure, the unified API of the [Javascript API ](https://github.com/getsentry/sentry-javascript/blob/d67ad66183649c8c09182ebaae773ad40aed532f/packages/hub/src/interfaces.ts#L7)is done in this way (the scope is `nullable` too) even though the [constructor](https://github.com/getsentry/sentry-javascript/blob/d67ad66183649c8c09182ebaae773ad40aed532f/packages/hub/src/hub.ts#L53) of the `Hub` does not allow passing `null` as scope, so it seems to be an inconsistency of the implementation. ping @HazAT as this code was originally written by himself

Reply:
Main Reason for allowing `null` Client is for to make testing easier. Not really necessary in PHP if we can mock it easily.

`Scope` is just optional which means if you do not pass an explicit `Scope` you created then the ctor will create a `new Scope` for you.

Reply:
As we can easily mock the client thanks for its interface and a layer does not makes sense to exist without it, I will change the constructor definition to `__construct(ClientInterface $client, ?Scope $scope = null)`